packages/grid/src/GridUtils.ts

```diff
@@ -51,6 +62,42 @@ export class GridUtils {
     return { x, y, row, column };
   }
 
+  static getCellInfoFromXY(
+    x: Coordinate,
+    y: Coordinate,
+    metrics: GridMetrics
+  ): CellInfo {
+    const { row, column } = GridUtils.getGridPointFromXY(x, y, metrics);
+
+    const {
+      visibleColumnWidths,
+      visibleRowHeights,
+      visibleColumnXs,
+      visibleRowYs,
+      modelColumns,
+      modelRows,
+    } = metrics;
+
+    const modelRow = row !== null ? modelRows.get(row) : null;
```

Comment:
Is the `!== null` check necessary? `null` is never going to be in these maps, could just rewrite all of these as (I think):
```
const modelRow = modelRows.get(row) ?? null;
```